src/svm/eventsClient.ts

```diff
@@ -0,0 +1,214 @@
+import { getDeployedAddress, SvmSpokeIdl } from "@across-protocol/contracts";
+import { getSolanaChainId } from "@across-protocol/contracts/dist/src/svm/web3-v1";
+import { BorshEventCoder, Idl, utils } from "@coral-xyz/anchor";
+import web3, {
+  Address,
+  Commitment,
+  GetSignaturesForAddressApi,
+  GetTransactionApi,
+  RpcTransport,
+  Signature,
+  unixTimestamp,
+} from "@solana/web3-v2.js";
+import { EventData, EventName, EventWithData } from "./types";
+import { getEventName, parseEventData } from "./utils/events";
+import { isDevnet } from "./utils/helpers";
+
+type GetTransactionReturnType = ReturnType<GetTransactionApi["getTransaction"]>;
+type GetSignaturesForAddressConfig = Parameters<GetSignaturesForAddressApi["getSignaturesForAddress"]>[1];
+type GetSignaturesForAddressTransaction = ReturnType<GetSignaturesForAddressApi["getSignaturesForAddress"]>[number];
+type GetSignaturesForAddressApiResponse = readonly GetSignaturesForAddressTransaction[];
+
+export class SvmSpokeEventsClient {
+  private rpc: web3.Rpc<web3.SolanaRpcApiFromTransport<RpcTransport>>;
+  private svmSpokeAddress: Address;
+
+  /**
+   * Private constructor. Use the async create() method to instantiate.
+   */
+  private constructor(rpc: web3.Rpc<web3.SolanaRpcApiFromTransport<RpcTransport>>, svmSpokeAddress: Address) {
+    this.rpc = rpc;
+    this.svmSpokeAddress = svmSpokeAddress;
+  }
+
+  /**
+   * Factory method to asynchronously create an instance of SvmSpokeEventsClient.
+   */
+  public static async create(
+    rpc: web3.Rpc<web3.SolanaRpcApiFromTransport<RpcTransport>>
+  ): Promise<SvmSpokeEventsClient> {
+    const isTestnet = await isDevnet(rpc);
+    const programId = getDeployedAddress("SvmSpoke", getSolanaChainId(isTestnet ? "devnet" : "mainnet").toString());
+    if (!programId) throw new Error("Program not found");
+    return new SvmSpokeEventsClient(rpc, web3.address(programId));
+  }
+
+  /**
+   * Queries events for the SvmSpoke program filtered by event name.
+   *
+   * @param eventName - The name of the event to filter by.
+   * @param fromSlot - Optional starting slot.
+   * @param toSlot - Optional ending slot.
+   * @param options - Options for fetching signatures.
+   * @param finality - Commitment level.
+   * @returns A promise that resolves to an array of events matching the eventName.
+   */
+  public async queryEvents<T extends EventData>(
+    eventName: EventName,
+    fromSlot?: bigint,
+    toSlot?: bigint,
+    options: GetSignaturesForAddressConfig = { limit: 1000 },
+    finality: Commitment = "confirmed"
```

Comment:
does it make sense to pass separate commitment in the `finality` parameter (used in `getTransaction` calls) from `options.commitment` that gets used in `getSignaturesForAddress` calls?